### PR TITLE
Blog app - Views

### PR DESCRIPTION
**Pull request by AnsarIbrahim**
# This PR Contains:

## Project Requirements

- Implement the design from the sneak peek wireframes✅
- Blog App - processing data in models✅
- Basic styling using Tailwind CSS to match the wireframes✅

### CODE REVIEW

File: app/views/users/show.html.erb [hunk 1]
```diff
@@ -1 +1,54 @@
-<h1>Hello this is what to show</h1>
+<h1 class="mt-5 font-bold">User Details</h1>
+
+<div class="mt-3">
```
Comment (topeogunleye) [inline app/views/users/show.html.erb:3]: Hello, and thank you for your submission! 😊 It looks like you have put a lot of effort into your project, and it's well appreciated. After reviewing the provided code against the project requirements, here are a few points to consider.

#### Observations:
- Your code is well-structured and clean, making it easier to review.
- I see you've managed to render the user details and posts quite well.

#### Points for Consideration:
- **Project Requirement Fulfillment:** The project requirement specifies the use of the `#recent_post` method to render the user's three most recent posts, but it seems like your implementation does not use this method and instead, iterates over all posts related to the user with `@user_lists.posts.each do |post|`. Please consider adjusting your code to meet the specified requirements. 💡

  ```erb
  <% @user.recent_post.each do |post| %>
  ...
  <% end %>
  ```

#### Suggestions:
- **Refactoring:** Consider refactoring the code to display only the three most recent posts as specified in the requirements.
  
#### Closing Thoughts:
Overall, you're on the right track! 🌟 A few modifications to align the project more closely with the requirements will make it even better. Please make the necessary adjustments and resubmit your pull request for review.

Thank you, and happy coding! 🚀

**Review by topeogunleye — CHANGES_REQUESTED** [review]
Hi @AnsarIbrahim, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

Try-Again

> ## (Highlights) Good Points: 💪 

#### 🌐 **Controllers and Views**
- Your implementation of the `UsersController#index` action correctly fetches all the users and renders them in the corresponding view. 🎉
- The `index.html.erb` is well-structured, efficiently iterating over the `@users` instance variable to display user details. 🏆

#### 🔄 **Rendering & Navigation**
- The navigation links for each user are working perfectly, redirecting to the appropriate show pages seamlessly, adhering to the wireframes provided. 🛣️
- The “See all Post” links are functional and navigate to the user's post index as illustrated in the wireframe. 🚀

#### 💬 **Comments & Posts Display**
- You successfully used `post.recent_comments` to render the 5 most recent comments in the `posts/index.html.erb` page. 💬
- The post show page effectively displays the post with all comments listed, aligning with the wireframe. 📄

#### 🌈 **Styling (Nice to Have)**
- Though the focus was on functionality, your attention to styling and detail is commendable and contributes to an improved user experience. 🎨

#### 🎁 **Extra Credit**
- 🏅 **Extra Credit**: You had the `app/views/users/_user.html.erb` partial, it's a more refined approach, earning you extra kudos! 🎁

#### 🌟 **Conclusion**
In conclusion, while you’ve made substantial progress and shown an understanding of the project requirements, there are some pivotal alterations and enhancements that are needed to fully align the project with the given specifications and best practices. 🛠️

The changes requested are crucial to refining your implementation and achieving a more polished and optimized outcome. Please review the specific comments and suggestions provided and make the necessary adjustments. 🔄

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

### There was no chance to put the below comments inline:

You can also consider:
- Consider populating your `db/seed.rb` file with data. Here's how:
```erb
first_user = User.create(name: 'Tom', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Mexico.')
second_user = User.create(name: 'Lilly', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Poland.')
first_post = Post.create(author: first_user, title: 'Hello', text: 'This is my first post')
Comment.create(post: first_post, user: second_user, text: 'Hi Tom!' )
```
- **Tailwind CSS Instructions:**
   Please include clear instructions in the README on running the Tailwind CSS watch command for development. 📚 

   ```markdown
   ### Running Tailwind CSS in Watch Mode
   
   ```sh
   rails tailwindcss:watch
   ```
   or 
   ```sh
   ./bin/dev
   ```

After addressing these suggested improvements and any other highlighted issues, kindly resubmit the project for another review. I believe with these modifications, your project will not only meet the expectations but excel in terms of quality and adherence to best practices! 🌟

Remember, continuous improvement is key in coding. Keep up the great work, and feel free to ask for clarifications or additional support if needed. Happy coding! 🚀
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/users/show.html.erb (hunk 1 above)
Comment (AnsarIbrahim) [reply app/views/users/show.html.erb:3]: Thanks for that and i change

Comment (topeogunleye) [reply app/views/users/show.html.erb:3]: you're welcome @AnsarIbrahim

**Review by KayLemba — APPROVED** [review]
[[# STATUS: APPROVED :green_circle: :clap: 

Hello @AnsarIbrahim    , 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  . 
Congratulations! 🎉  Keep up the Awesome work 🥂  Keep Rocking 🥇 

HIGHLIGHTS:

- [x]  Your repo is professional  😎 
- [x] Descriptive P.R 💯 
- [x] Professional README file 
- [x] App running as intended. 💯 🥇 




 

 










 








Now you can merge this branch and move on :1st_place_medal: 
Happy Coding cheers 🥇 💯 



well-1

Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.



Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._]]

**Pull request merged**